src/blob.rs

```diff
@@ -349,6 +353,31 @@ impl<'a> BlobObject<'a> {
         }
         true
     }
+
+    pub fn recode_to_avatar_size(&self, context: &Context) -> Result<(), BlobError> {
+        let blob_abs = self.to_abs_path();
+        let img = image::open(&blob_abs).map_err(|err| BlobError::RecodeFailure {
+            blobdir: context.get_blobdir().to_path_buf(),
+            blobname: blob_abs.to_str().unwrap_or_default().to_string(),
+            cause: err,
+            backtrace: failure::Backtrace::new(),
+        })?;
```

Comment:
i am wondering if this can be more elegant, just bubble up an ImageError without thinking much about it.